### ISSUE TITLE
new version of letsencrypt v1.5.0

### ISSUE THREAD

**Issue by coolaj86** [spacekit/spacekit-service#35]
Just a heads up:

I just published a new version of letsencrypt - v1.5.0

There shouldn't be any breaking changes to the API, but it does require a shim for the new Buffer methods that aren't implemented in older version of node.

Let me know if you have any problems or if I can answer any questions.

**Comment by jedireza** [comment]
@coolaj86 in node@v6 things are fine but our travis build with node@v4 bombs out with:

``` bash
/home/travis/build/spacekit/spacekit-service/node_modules/letsencrypt/node_modules/rsa-compat/node_modules/buffer-v6-polyfill/index.js:19
Buffer.from = newBuffer;
            ^
TypeError: Cannot assign to read only property 'from' of function Buffer(arg, encoding) {
```

Thoughts?

**Comment by coolaj86** [comment]
Reinstall now. I just updated `buffer-v6-polyfill` to not overwrite `from` if it exists.

Buffer in v6 has introduced breaking change behavior (deprecating the use of `new Buffer()`) which requires a shim to be backwards compatible in v4.

**Comment by jedireza** [comment]
Works like a charm. Thanks for the quick response. 🙏

**Comment by coolaj86** [comment]
As another heads up I got on a roll and started fixing a bunch of bugs and adding requested features. 

The result will soon be available as v2.0.0, but the docs are available now:

https://github.com/Daplie/node-letsencrypt/tree/v2.x

I'm still in the process of stripping it down before publishing, but it's already much, much simpler.

**Comment by coolaj86** [comment]
v2 is out now: https://github.com/Daplie/node-letsencrypt
- Smaller code base
- More thorough tests
- Hooks for saving to sql, mongo, etc (plugin architecture for storage)
- All bugs squashed (at least I'm pretty sure)
- Supports multiple altnames
- Better windows support (fixed filepath issues in `le-store-certbot`)

You can see the new API examples in the README and in the tests:
https://github.com/Daplie/node-letsencrypt/tree/master/tests
